torch/fx/experimental/proxy_tensor.py

```diff
@@ -776,7 +878,12 @@ def wrapped(*args):
         from .symbolic_shapes import ShapeEnv
 
         phs = pytree.tree_map(lambda _: fx.PH, args)  # type: ignore[attr-defined]
-        fx_tracer = PythonKeyTracer()
+
+        if hasattr(f, "_orig_mod"):
+            scope_root = f._orig_mod
+            fx_tracer = _ModuleStackTracer(scope_root)
+        else:
+            fx_tracer = PythonKeyTracer()
```

Comment:
Seems a little sketch, the way I thought you would have done this is: 1) add a config for make_fx to trace nn_module_stack, 2) turn the config on when we're inside export. This makes the behavior explicit (instead of needing to depend on if a special attr has been set on a module)

Reply:
@bdhirsh what do you think? We did it this way because we didn't want to plumb more configs in aot_autograd as the flag and the module had to be propagated down from aot_export module to make_fx (going through like 3, 4 layers)

Reply:
@zou3519 I'm onboard with being more explicit. One annoyance is that `make_fx` needs the root module to track stacktraces properly, which will require plumbing [this nn module](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/traced_function_transforms.py#L595) all the way down to the `make_fx` call in AOTAutograd (currently it gets wrapped by several layers of wrapper functions that we trace through).

Any preference for these 3 options?
(1) PR as is (very implicit)
(2) Keep stashing `_orig_mod` somewhere, but add a kwarg to `make_fx`, `maintain_module_stack=True`, that instructs make_fx to actually use `_orig_mod` and perform the bookkeeping. If that config is set, we would probaby error if `_orig_mod` doesn't exist and the traced function is not an `nn.Module`.
(3) Plumb the original nn module through AOTAutograd and have it be an explicit argument to `make_fx`

Typing these out, I think I prefer (2) (avoids the extra plumbing, more explicit, and also lets us raise an error if didn't set / plumb `_orig_mod` properly)

Reply:
Yeahh it seems (2) is better for me.